test/integ/pandas/test_pandas_tools.py

```diff
@@ -315,7 +315,7 @@ def test_empty_dataframe_write_pandas(
 
 
 @pytest.mark.parametrize("quote_identifiers", [True, False])
-def test_location_building_db_schema(conn_cnx, quote_identifiers: bool):
+def test_table_location_building_db_schema(conn_cnx, quote_identifiers: bool):
```

Comment:
Can we combine the three cases (`db_schema`, `schema`, neither specified) into one using `parametrize`? Maybe something like this:
```
@pytest.mark.parametrize("database,schema,quote_identifiers,expected_location", [
    ("database", "schema", True, "\"database\".\"schema\".\"table\""),
    ("database", "schema", False, "database.schema.table"),
    (None, "schema", True, "\"schema\".\"table\""),
    (None, "schema", False, "schema.table"),
    (None, None, True, "\"table\""),
    (None, None, False, "table")
])
def test_table_location_building(conn_cnx, database:str|None, schema:str|None, quote_identifiers: bool, expected_location:str):
    """This tests that write_pandas constructs location correctly with database, schema and table name."""
    from snowflake.connector.cursor import SnowflakeCursor

    with conn_cnx() as cnx:
        def mocked_execute(*args, **kwargs):
            if len(args) >= 1 and args[0].startswith("COPY INTO"):
                location = args[0].split(" ")[2]
                assert location == expected_location
            cur = SnowflakeCursor(cnx)
            cur._result = iter([])
            return cur

        with mock.patch(
            "snowflake.connector.cursor.SnowflakeCursor.execute",
            side_effect=mocked_execute,
        ) as m_execute:
            success, nchunks, nrows, _ = write_pandas(
                cnx,
                sf_connector_version_df.get(),
                "table",
                database=database,
                schema=schema,
                quote_identifiers=quote_identifiers,
            )
            assert m_execute.called and any(
                map(lambda e: "COPY INTO" in str(e[0]), m_execute.call_args_list))
``` 
What do you think?

Reply:
Same applies for `stage_location` and `file_format_location` tests.

Reply:
Like this idea, makes it much more clear and easier to understand. Did it [here](https://github.com/snowflakedb/snowflake-connector-python/pull/1274/commits/331b999014708427e25e8d11a3ce0aea5642cb58) for 3 of them.